src/test/java/software/amazon/encryption/s3/S3EncryptionClientStreamTest.java

```diff
@@ -203,6 +203,69 @@ public void ordinaryInputStreamV3DecryptCbc() throws IOException {
         v3Client.close();
     }
 
+    @Test
+    public void invalidBufferSize() {
+        assertThrows(S3EncryptionClientException.class, () -> S3EncryptionClient.builder()
+                .kmsKeyId(KMS_KEY_ID)
+                .maxBufferSize(16)
+                .build());
+        assertThrows(S3EncryptionClientException.class, () -> S3EncryptionClient.builder()
+                .kmsKeyId(KMS_KEY_ID)
+                .maxBufferSize(4096)
+                .build());
+    }
+
+    @Test
+    public void customMaxBufferSizeWithLargeObject() throws IOException {
+        final String objectKey = appendTestSuffix("large-object-test");
+
+        Security.addProvider(new BouncyCastleProvider());
+        Provider provider = Security.getProvider("BC");
+
+        // V3 Client with custom max buffer size 32 MiB.
+        S3Client v3ClientWithBuffer32MiB = S3EncryptionClient.builder()
+                .aesKey(AES_KEY)
+                .cryptoProvider(provider)
+                .maxBufferSize(32)
+                .build();
+
+        // V3 Client with default buffer size (i.e. 64MiB)
+        S3Client v3ClientWithDelayedAuth = S3EncryptionClient.builder()
+                .aesKey(AES_KEY)
+                .cryptoProvider(provider)
+                .maxBufferSize(32)
+                .enableDelayedAuthenticationMode(true)
```

Comment:
We should validate against this case. When `enableDelayedAuthenticationMode` is set to `true` we don't buffer anything, so it shouldn't be possible to configure `maxBufferSize`.

Reply:
agreed.

Reply:
Just to confirm, we shouldn't throw an exception even though both `maxBufferSize` and `enableDelayedAuthenticationMode` set to true. But, if `enableDelayedAuthenticationMode` is enabled, this mode always takes priority over Buffered Mode.

Reply:
If `enableDelayedAuthenticationMode` is set to `true` AND `maxBufferSize` is set, we should throw an exception. It shouldn't be possible to configure the buffer size, when there is no buffer to configure.